DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -7,201 +7,324 @@
+              self.rmsMonitoringReporter.addRecord({
+                  "timestamp": time.mktime(datetime.datetime.utcnow().timetuple()),
+                  "host": socket.getfqdn(),
+                  "objectType": "File",
+                  "operationType": self.operation.Type,
+                  "objectID": opFile.FileID,
+                  "parentID": self.operation.OperationID,
+                  "status": status,
+                  "nbObject": 1
+              })
+          self.rmsMonitoringReporter.commit()
+        else:
+          gMonitor.addMark("ReplicateAndRegisterAtt", len(self.operation))
+          gMonitor.addMark("ReplicateFail", len(self.operation))
         return bannedSource
 
       if bannedSource["Value"]:
         self.operation.Error = "SourceSE %s is banned for reading" % sourceSE
-        self.log.info( self.operation.Error )
-        return S_OK( self.operation.Error )
+        self.log.info(self.operation.Error)
+        return S_OK(self.operation.Error)
 
     # # check targetSEs for write
     bannedTargets = self.checkSEsRSS()
     if not bannedTargets['OK']:
-      gMonitor.addMark( "ReplicateAndRegisterAtt", len( self.operation ) )
-      gMonitor.addMark( "ReplicateFail", len( self.operation ) )
+      if self.rmsMonitoring:
+        for opFile in self.operation:
+          for status in ["FileAttempted", "FileFailed"]:
+            self.rmsMonitoringReporter.addRecord({
+                "timestamp": time.mktime(datetime.datetime.utcnow().timetuple()),
+                "host": socket.getfqdn(),
+                "objectType": "File",
+                "operationType": self.operation.Type,
+                "objectID": opFile.FileID,
+                "parentID": self.operation.OperationID,
+                "status": status,
+                "nbObject": 1
+            })
+        self.rmsMonitoringReporter.commit()
+      else:
+        gMonitor.addMark("ReplicateAndRegisterAtt", len(self.operation))
+        gMonitor.addMark("ReplicateFail", len(self.operation))
       return bannedTargets
 
     if bannedTargets['Value']:
-      self.operation.Error = "%s targets are banned for writing" % ",".join( bannedTargets['Value'] )
-      return S_OK( self.operation.Error )
+      self.operation.Error = "%s targets are banned for writing" % ",".join(bannedTargets['Value'])
+      return S_OK(self.operation.Error)
 
     # Can continue now
-    self.log.verbose( "No targets banned for writing" )
+    self.log.verbose("No targets banned for writing")
 
     # # check sourceSEs for removal
     # # for removal the targetSEs are the sourceSEs of the replication
     targetSEs = self.operation.sourceSEList
-    bannedTargets = self.checkSEsRSS( targetSEs, access = 'RemoveAccess' )
+    bannedTargets = self.checkSEsRSS(targetSEs, access='RemoveAccess')
     if not bannedTargets['OK']:
-      gMonitor.addMark( "RemoveReplicaAtt" )
-      gMonitor.addMark( "RemoveReplicaFail" )
+      if self.rmsMonitoring:
```

Comment:
same as before ( I won't repeat, I let you find and check the other occurrences)